### ISSUE TITLE
Stacking Fertilized Dirt blocks in world causes server crash

### ISSUE THREAD

**Issue by adkinss** [LordDeatHunter/Croptosis#2]
Version: Croptosis 1.0.1
API: Fabric API 0.30.3+1.16
ModPack: All of Fabric 3 v2.6.9

I wanted to reach a beehive higher up in a tree and started placing dirt blocks, one on top of the other, to reach the hive.  I got a handful of blocks up and the server crashed.  Server is now in an endless crash loop.  I realized after the crash that I was using fertilized farmland as the dirt blocks and not regular dirt blocks.

Place fertilized farmland in world, one on top of the other.  It doesn't take long to crash the server.

In the chunk data, I see **croptosis:fertilized_dirt** with property **max_height: 3**.

Here is the stack trace:
> java.lang.StackOverflowError: Exception ticking world
>         at net.minecraft.class_1937.getChunkLithium(class_1937.java:5740)
>         at net.minecraft.class_1937.method_8497(class_1937.java:5726)
>         at net.minecraft.class_1937.method_8320(class_1937.java:408)
>         at wraith.croptosis.block.FertilizedDirtBlock.method_9514(FertilizedDirtBlock.java:38)
>         at wraith.croptosis.block.FertilizedDirtBlock.method_9514(FertilizedDirtBlock.java:52)
>         at wraith.croptosis.block.FertilizedDirtBlock.method_9514(FertilizedDirtBlock.java:52)
>         at wraith.croptosis.block.FertilizedDirtBlock.method_9514(FertilizedDirtBlock.java:52)
>         at wraith.croptosis.block.FertilizedDirtBlock.method_9514(FertilizedDirtBlock.java:52)
>         at wraith.croptosis.block.FertilizedDirtBlock.method_9514(FertilizedDirtBlock.java:52)
>         ....

**Comment by adkinss** [comment]
I replaced croptosis:fertilized_dirt with minecraft:dirt in the chunk and was able to get past my crash issue.  Looking at the code, I believe the problem is that an assumption is being made that the block above the fertilized dirt is able to be ticked and the randomTick() is called on it.  If a block can't be ticked, the potential for a crash exists.

**Comment by adkinss** [comment]
I was able to reproduce the crash bug in a single player world with only the Croptosis mod installed (no modpack, just this one mod).  The crash is triggered when the top-most block that is randomly ticked is also a Fertilized Dirt block or a Fertilized Sand block, resulting in recursive calls to randomTick().  I wasn't able to determine what triggers the endless recursion loop, which eventually overflows the stack and crashes the game.

To fix it, I refactored the loop to check the top most block and randomTick() it if it is not a Fertilized Dirt or Sand.  I will submit a PR for your review.
